src/overlays/actors/ovl_En_Wallmas/z_en_wallmas.c

```diff
@@ -43,147 +55,702 @@ const ActorInit En_Wallmas_InitVars = {
+        this->collider.base.colType = 0;
+        this->drawDmgEffAlpha = 0.0f;
+        Actor_SpawnIceEffects(globalCtx, &this->actor, this->limbPos, 11, 2, 0.3f, 0.2f);
+        this->actor.flags |= ACTOR_FLAG_400;
+    }
+}
+
+void EnWallmas_TimerInit(EnWallmas* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actor.flags |= ACTOR_FLAG_20;
+    this->timer = 130;
+    this->actor.velocity.y = 0.0f;
+    this->actor.world.pos.y = player->actor.world.pos.y;
+    this->actor.floorHeight = player->actor.floorHeight;
+    this->actor.draw = EnWallmas_Draw;
+    this->actionFunc = EnWallmas_WaitToDrop;
+}
+
+void EnWallmas_WaitToDrop(EnWallmas* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f* playerPos = &player->actor.world.pos;
+
+    this->actor.world.pos = *playerPos;
+    this->actor.floorHeight = player->actor.floorHeight;
+    this->actor.floorPoly = player->actor.floorPoly;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if ((player->stateFlags1 & 0x08100000) || (player->stateFlags2 & 0x80) || (player->unk_B5E > 0) ||
+        (player->actor.freezeTimer > 0) || !(player->actor.bgCheckFlags & 1) ||
+        ((EN_WALLMAS_GET_TYPE(&this->actor) == WMT_PROXIMITY) &&
+         (Math_Vec3f_DistXZ(&this->actor.home.pos, playerPos) > (120.f + this->detectionRadius)))) {
+        func_801A75E8(NA_SE_EN_FALL_AIM);
+        this->timer = 130;
+    }
+
+    if (this->timer == 80) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_FALL_AIM);
+    }
+
+    if (this->timer == 0) {
+        EnWallmas_SetupDrop(this, globalCtx);
+    }
+}
+
+void EnWallmas_SetupDrop(EnWallmas* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    Animation_Change(&this->skelAnime, &gWallmasterLungeAnim, 0.0f, 20.0f,
+                     Animation_GetLastFrame(&gWallmasterLungeAnim), ANIMMODE_ONCE, 0.0f);
+
+    this->yTarget = player->actor.world.pos.y;
+    this->actor.world.pos.y = player->actor.world.pos.y + 300.0f;
+    this->actor.shape.rot.y = player->actor.shape.rot.y + 0x8000;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actor.floorHeight = player->actor.floorHeight;
+    this->actor.flags |= ACTOR_FLAG_1;
+    this->actor.flags &= ~ACTOR_FLAG_20;
+    this->actionFunc = EnWallmas_Drop;
+}
+
+void EnWallmas_Drop(EnWallmas* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if ((player->stateFlags2 & 0x80) || (player->actor.freezeTimer > 0)) {
+        EnWallmas_SetupReturnToCeiling(this);
+    } else if (!func_801690CC(globalCtx) && !(player->stateFlags2 & 0x10) && (player->invincibilityTimer >= 0) &&
+               (this->actor.xzDistToPlayer < 30.0f) && (this->actor.playerHeightRel < -5.0f) &&
+               (-(f32)(player->cylinder.dim.height + 10) < this->actor.playerHeightRel)) {
```

Comment:
I assume this is required to account for the float operations in some way?

Reply:
Yeah that `(f32)` cast is required here. Here's what it looks like without:
![image](https://user-images.githubusercontent.com/12245827/159151640-9851dcf4-6e8d-4281-a59c-59a957af8b26.png)